websocket/factory/factory.go

```diff
@@ -0,0 +1,82 @@
+package factory
+
+import (
+	"github.com/multiversx/mx-chain-communication-go/websocket"
+	"github.com/multiversx/mx-chain-communication-go/websocket/client"
+	outportData "github.com/multiversx/mx-chain-communication-go/websocket/data"
+	"github.com/multiversx/mx-chain-communication-go/websocket/driver"
+	"github.com/multiversx/mx-chain-communication-go/websocket/server"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+)
+
+// ArgsWebSocketDriverFactory holds the arguments needed for creating a webSocketsDriverFactory
+type ArgsWebSocketDriverFactory struct {
+	WebSocketConfig          outportData.WebSocketConfig
+	Marshaller               marshal.Marshalizer
+	Uint64ByteSliceConverter websocket.Uint64ByteSliceConverter
+	Log                      core.Logger
+}
+
+// NewWebSocketDriver will handle the creation of all the components needed to create an outport driver that sends data over WebSocket
+func NewWebSocketDriver(args ArgsWebSocketDriverFactory) (websocket.Driver, error) {
```

Comment:
Instead of driver, should we switch to Host? I think `websocket.Host` sounds better. The method will then become `NewWebSocketHost` and the argument `ArgsWebSocketHost`

Reply:
will be refactor in the next PR